File: src/main/java/com/sprint/BookPartnerApplication/services/StoreService.java
package com.sprint.BookPartnerApplication.services;

import com.sprint.BookPartnerApplication.dto.response.SalesResponseDTO;
import com.sprint.BookPartnerApplication.dto.request.StoreRequestDTO;
import com.sprint.BookPartnerApplication.dto.response.StoreResponseDTO;
import com.sprint.BookPartnerApplication.entity.Discounts;
import java.util.List;

public interface StoreService {

    StoreResponseDTO createStore(StoreRequestDTO dto);
    List<StoreResponseDTO> getAllStores();
    StoreResponseDTO getStoreById(String storeId);
    StoreResponseDTO updateStore(String storeId, StoreRequestDTO dto);
    List<SalesResponseDTO> getSalesByStore(String storeId);
    List<Discounts> getDiscountsByStore(String storeId);
}

File: BookPartner-UI/src/app/service/store.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class StoreService {
  private baseUrl = 'http://localhost:8082/api/v1/stores';

  constructor(private http: HttpClient) {}

  createStore(dto: any): Observable<any> {
    return this.http.post(`${this.baseUrl}`, dto);
  }

  getAllStores(): Observable<any[]> {
    return this.http.get<any[]>(`${this.baseUrl}`);
  }

  getStoreById(storeId: string): Observable<any> {
    return this.http.get(`${this.baseUrl}/${storeId}`);
  }

  updateStore(storeId: string, dto: any): Observable<any> {
    return this.http.put(`${this.baseUrl}/${storeId}`, dto);
  }

  getSalesByStore(storeId: string): Observable<any[]> {
    return this.http.get<any[]>(`${this.baseUrl}/${storeId}/sales`);
  }

  getDiscountsByStore(storeId: string): Observable<any[]> {
    return this.http.get<any[]>(`${this.baseUrl}/${storeId}/discounts`);
  }
}